sys-block/gpart/gpart-0.3.recipe

```diff
@@ -0,0 +1,49 @@
+SUMMARY="A tool for recovering MBR partition tables in hard disks"
+DESCRIPTION="Gpart is a small tool which tries to guess what partitions are \
+on a PC type, MBR-partitioned hard disk in case the primary partition table \
+was damaged."
+HOMEPAGE="https://github.com/baruch/gpart"
+COPYRIGHT="1999-2017 Michail Brzitwa"
+LICENSE="GNU GPL v2"
+REVISION="1"
+srcGitRev="23b651d1e9c6ae1454e5d3fe315d7d4e145661c0"
+SOURCE_URI="https://github.com/baruch/gpart/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="c492b7dc677e4c6af197f9598ba815a4e14a03578ee8a963430529767bc6c947"
+SOURCE_FILENAME="gpart-$portVersion-$srcGitRev.tar.gz"
+SOURCE_DIR="gpart-$srcGitRev"
+PATCHES="gpart-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 ?x86 x86_64"
+
+PROVIDES="
+	gpart = $portVersion
+	cmd:gpart = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:autoconf
+	cmd:automake
+	cmd:gcc
+	cmd:make
+	"
+
+defineDebugInfoPackage gpart \
+	$binDir/gpart
```

Comment:
On x86_gcc2, it appears to me that `gpart` was built without the `-g` flags. This means there won't be any (useful) debuginfos available in the binary, so this should be removed

```diff
-defineDebugInfoPackage gpart \
-	$binDir/gpart

Reply:
@alaviss, you had convinced me about the fact that an empty debuginfo package is worse than no debugpackage at all, but I wonder if it would not be, after all, better to keep these empty debuginfo packages and, as you suggested during that discussion, teach haikuporter how to trigger a debug build.
So, maybe we could keep the defineDebugInfoPackage in the hope that haikuporter gains this feature.
What do you think?

Reply:
How would HaikuPorter do that? Making debug builds varies with each port, there isn't really a way to standardize or factor this out. The recipe needs to take care of this. If generally enabling debug builds is undesireable for some reason (performance for example), then there should just not be any debuginfo package. The recipe could also build the binaries twice and provide a debug build as a separate subpackage, but the user would then explicitly have to install that one. Making a debuginfo package from a separate debug build wouldn't work however, as the info wouldn't match the non-debug build.

Reply:
I might be wrong, but I **_think_** that with a **_very dirty_** hack like asking haikuporter to define a `HAIKU_CFLAGS="-g"` and patching our gcc to examine HAIKU_CFLAGS it would be possible to have non fake debuginfo packages, and this would probably work for many recipes.
There might even be less ugly solutions.

But @alaviss and @pulkomandy have already convinced me that it is not a good idea to have fake debuginfo packages.

BTW, @mmlr, I'm just replying here because you asked how I would do that. But I don't even plan to try that :)

Reply:
@fbrosson we don't have to patch gcc for that. Simply appending the flags to our CFLAGS is fine enough.

@mmlr There are two kinds of debug builds:
- The one where you get a binary with all debugging logic on (ie. `ASSERT()`): For this, it would have to be done by the port as different build system have different ways to enable debugging logic
- The one where you only have good backtrace and step-by-step debugging: For this all you need is to make the compiler generate debugging infomations, mainly by appending `-g` to CFLAGS.

Gentoo has an [excellent write up][0] on this topic, and I would recommend checking it out.

[0]: https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Backtraces

Reply:
Yes, I know about the different debug flags. Also our `defineDebugInfoPackage` basically is what `splitdebug` does in gentoo.

But messing with compiler flags is still just not the job of HaikuPorter, it is the job of the recipe. Only the individual recipe really knows / should know what the software in question needs. Maybe the software isn't even compiled with a compiler that understands certain flags, or it isn't compiled at all.

Having such setup done locally by the recipe is just a lot more obvious and can be tailored to the actual situation.